config/config.go

```diff
@@ -35,18 +37,58 @@ var (
 	ErrRemoteConfigNotFound  = errors.New("remote config not found")
 	ErrRemoteConfigEmptyURL  = errors.New("remote config: empty URL")
 	ErrRemoteConfigEmptyName = errors.New("remote config: empty name")
+
+	// ActiveFS provides a configurable entry point to the billy.Filesystem in active use for reading global and
+	// system level configuration; override in a test to use a fake filesystem then reset to config.DefaultFS() to
+	// restore the default behavior.
+	ActiveFS = DefaultFS()
 )
 
+// DefaultFS provides a billy.Filesystem abstraction over the os filesystem (via osfs.OS) scoped to the
+// root directory / in order to enable access to global and system config files via absolute paths
+func DefaultFS() billy.Filesystem {
+	return osfs.New("/")
+}
+
 // Scope defines the scope of a config file, such as local, global or system.
 type Scope int
 
-// Available ConfigScope's
+// Available ConfigScopes; those with a V6 prefix are opt-ins to preview go-git v6 config behavior which
+// more closely matches the `git config` subcommand documentation
 const (
 	LocalScope Scope = iota
 	GlobalScope
 	SystemScope
+
+	// V6DefaultScope is equivalent to running config commands without any explicit flags
+	V6DefaultScope
+
+	// V6SystemScope is equivalent to running config commands with the --system flag
+	V6SystemScope
+
+	// V6GlobalScope is equivalent to running config commands with the --global flag
+	V6GlobalScope
+
+	// V6LocalScope is equivalent to running config commands with the --local flag
+	V6LocalScope
```

Comment:
What are your thoughts on adding `V6` as a suffix instead of a prefix?

Also, what is the transition/migration strategy for the V6 release? Would we keep the `V6` suffixed Scopes and remove the previous ones?

It would be good to add a TODO for v6 on the outstanding steps.

Reply:
Moving V6 to a suffix is a fine choice.

As for the transition/migration strategy, I would leave that to you and the maintaining team.

My recommendation from a client/consumer perspective would be to

1. have go-git v5 continue to default to the existing behavior with v6 as an opt-in choice
2. with the release of go-git v6 announce this as a breaking change, flip the default behavior to v6, and either remove or leave the v5 behavior as an opt-in choice for legacy compatibility. in this case, I suggest announcing that it will be maintained for backward compatibility until v7 (or some other point in the future) but is deprecated and no longer supported as the recommended support path is to update to the v6 behavior.